Kernel/Net/TCPSocket.cpp

```diff
@@ -414,19 +415,45 @@ NetworkOrdered<u16> TCPSocket::compute_tcp_checksum(IPv4Address const& source, I
 
 ErrorOr<void> TCPSocket::protocol_bind()
 {
-    return m_adapter.with([this](auto& adapter) -> ErrorOr<void> {
+    dbgln_if(TCP_SOCKET_DEBUG, "TCPSocket::protocol_bind(), local_port() is {}", local_port());
+    // Check that we do have the address we're trying to bind to.
+    TRY(m_adapter.with([this](auto& adapter) -> ErrorOr<void> {
         if (has_specific_local_address() && !adapter) {
             adapter = NetworkingManagement::the().from_ipv4_address(local_address());
             if (!adapter)
                 return set_so_error(EADDRNOTAVAIL);
         }
         return {};
-    });
-}
-
-ErrorOr<void> TCPSocket::protocol_listen(bool did_allocate_port)
-{
-    if (!did_allocate_port) {
+    }));
+
+    if (local_port() == 0) {
+        // Allocate an unused ephemeral port.
```

Comment:
I realize now that you only moved the below allocation code, but take at least the non-nitpicky comments into account :^)

Reply:
Yes, this is just moving the code from another place.

> but take at least the non-nitpicky comments into account :^)

So... do I understand it right that you want me to add another commit on top changing the `for` loop into a `while` loop?

Reply:
> So... do I understand it right that you want me to add another commit on top changing the `for` loop into a `while` loop?

I'd change this in the commit that restructures the allocation code, so at the same time as you're moving the code.